modules/va_forms/app/workers/va_forms/form_reloader.rb

```diff
@@ -8,8 +8,20 @@ class FormReloader
 
     BASE_URL = 'https://www.va.gov'
 
+    def initialize
+      @processed_forms = []
+    end
+
     def perform
       load_page(current_page: 0)
+      mark_stale_forms
+    end
+
+    def mark_stale_forms
+      processed_form_names = @processed_forms.map { |f| f['form_name'] }
+      missing_forms = VaForms::Form.where.not(form_name: processed_form_names)
+      # rubocop:disable Rails/SkipsModelValidations
+      missing_forms.update_all(valid_pdf: false)
```

Comment:
I need some help on this, because I know disabling rubocop is probably not the best thing to do. Basically, as this line suggests, I just need to update all of these forms to `valid_pdf: false`. If I loop through `missing_forms`, then do -

```
form.valid_pdf = false
form.save
``` 

The save fails. It seems to be failing validation, but I just pulled the forms during the `Form.where.not(`. So, why would that fail validation?

Reply:
Unfortunately update_all is way faster because it doesn't instantiate an active record object for each record and instead deals directly with the database. In your case, since you need validations and callbacks, you'll need to instantiate the objects and so you're best bet is iterating in batches of 1000 and performing the update as originally shown. Such as:

`VaForms::Form.where.not(form_name: processed_form_names).find_each { |u| u.update(valid_pdf: false) }`

Reply:
Thanks @bastosmichael! Updated ✅